changelog/pending/20230517--sdk-go--fix-regression-disallowing-placing-a-pulumi-program-in-a-subdirectory-of-a-module.yaml

```diff
@@ -0,0 +1,4 @@
+changes:
+- type: fix
+  scope: sdk/go
+  description: Fix regression disallowing placing a Pulumi program in a subdirectory of a Go module.
```

Comment:
Just a friendly reminder with love: if we're characterizing this as a regression, we need to write a postmortem for it. It sounds like it is indeed a regression.

Reply:
Good point. Will follow up in https://github.com/pulumi/home/issues/2872